rpms/s3/s3rpm.spec

```diff
@@ -428,13 +475,65 @@ rm -rf %{buildroot}
+    [ -f /opt/seagate/cortx/auth/resources/authserver.properties ] ||
+        cp /opt/seagate/cortx/auth/resources/authserver.properties.sample /opt/seagate/cortx/auth/resources/authserver.properties
+    [ -f /opt/seagate/cortx/auth/resources/keystore.properties ] ||
+        cp /opt/seagate/cortx/auth/resources/keystore.properties.sample /opt/seagate/cortx/auth/resources/keystore.properties
+    echo "[cortx-s3server-rpm] INFO: S3 RPM Post Install section completed"
+elif [ $1 == 2 ];then
+    echo "[cortx-s3server-rpm] INFO: S3 RPM Post Upgrade section started"
+    python3.6 /opt/seagate/cortx/s3/bin/merge.py
+    echo "[cortx-s3server-rpm] WARNING: All mini-provisioner template files are overwritten."
+    echo "[cortx-s3server-rpm] INFO: S3 RPM Post Upgrade section completed"
+fi
 systemctl daemon-reload
 systemctl enable s3authserver
 systemctl restart rsyslog
 openssl_version=`rpm -q --queryformat '%{VERSION}' openssl`
 if [ "$openssl_version" != "1.0.2k" ] && [ "$openssl_version" != "1.1.1" ]; then
-  echo "Warning: Unsupported (untested) openssl version [$openssl_version] is installed which may work."
-  echo "Supported openssl versions are [1.0.2k, 1.1.1]"
+  echo "[cortx-s3server-rpm] WARNING: Unsupported (untested) openssl version [$openssl_version] is installed which may work."
+  echo "[cortx-s3server-rpm] INFO: Supported openssl versions are [1.0.2k, 1.1.1]"
+fi
+echo "[cortx-s3server-rpm] INFO: S3 RPM Post section completed"
+
+
+################################
+# post uninstall/upgrade section
+################################
+%postun
+if [ $1 == 1 ];then
+    echo "[cortx-s3server-rpm] INFO: S3 RPM Post Uninstall Upgrade section started"
+    # removed temporary files from /tmp
+    rm -f /tmp/*.sample.old
+    echo "[cortx-s3server-rpm] INFO: Removed temporary .old files from /tmp"
+    echo "[cortx-s3server-rpm] INFO: S3 RPM Post Uninstall Upgrade section completed"
+elif [ $1 == 0 ];then
+    echo "[cortx-s3server-rpm] INFO: S3 RPM Post Uninstall section started"
+    # remove config files.
+    rm -f /opt/seagate/cortx/s3/conf/s3config.yaml*
+    rm -f /opt/seagate/cortx/s3/conf/s3config_unsafe_attributes.yaml
+    rm -f /opt/seagate/cortx/s3/s3backgrounddelete/config.yaml*
+    rm -f /opt/seagate/cortx/s3/s3backgrounddelete/s3backgrounddelete_unsafe_attributes.yaml
+    rm -f /opt/seagate/cortx/s3/s3backgrounddelete/s3_cluster.yaml*
+    rm -f /opt/seagate/cortx/s3/s3backgrounddelete/s3_cluster_unsafe_attributes.yaml
+    rm -f /opt/seagate/cortx/auth/resources/authserver.properties*
+    rm -f /opt/seagate/cortx/auth/resources/authserver_unsafe_attributes.properties
+    rm -f /opt/seagate/cortx/auth/resources/keystore.properties*
+    rm -f /opt/seagate/cortx/auth/resources/keystore_unsafe_attributes.properties
+    echo "[cortx-s3server-rpm] INFO: Removed all S3 config files"
+    echo "[cortx-s3server-rpm] INFO: S3 RPM Post Uninstall Upgrade section completed"
```

Comment:
Remove word upgrade from here

Reply:
this is to differentiate between actual post uninstall(rpm -e) and post uninstall during upgrade(rpm -Uvh)